### DISCUSSION TITLE
k3s server crashes abruptly when approximately 300 requests are sent in a span of 60 seconds (client QPS is  5 and bursty limit is 10)

### DISCUSSION THREAD

**Discussion by sharrurashmi** [k3s-io/k3s#10418] [Q&A]
**Environmental Info:**
K3s Version: 
1.30.1

Node(s) CPU architecture, OS, and Version: 
Linux  5.15.0-107-generic #117-Ubuntu SMP Fri Apr 26 12:26:49 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration:
one server one agent 

**Describe the bug:**
K3s server crashes when helm upgrade of a chart  is triggered , this basically means that the server crashes when sending around 300 api calls to the  server 

**Steps To Reproduce:**

- Installed K3s: 
- We have a helm chart installation of our product which was installed successfully 
- now when we start upgrading the helm chart we see that k3s server crashes occasionally 
- to the api server this means that the server crashes on handling around 300 requests ( the number of requests is calculated based on the resources we have in our cluster )
**Expected behavior:**
k3s server should not crash 

**Actual behavior:**
k3s server crashes 

**Additional context / logs:**

This is the stack trace 

```
18]: I0617 20:04:05.902329   37918 replica_set.go:676] "Finished syncing" kind="ReplicaSet" key="default/f5-csm-api-engine-5bc6f6b55" duration="167.85Âµs"
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime: g 16385: unexpected return pc for k8s.io/client-go/pkg/apis/clientauthentication.(*ExecCredential).XXX_DiscardUnknown called from 0x40
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: stack: frame={sp:0xc009ab1d90, fp:0xc009ab1db8} stack=[0xc009ab1000,0xc009ab2000)
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1c90:  0x000000c009ab1cc0  0x000000000293812f <go:(*struct { io.ReadCloser }).Read+0x000000000000002f>
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1ca0:  0x000000c0095681c0  0x000000c00a2d24e8
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1cb0:  0x0000000000000000  0x00007f0151ceca68
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1cc0:  0x000000c009ab1d08  0x00000000004b4efa <io.ReadAtLeast+0x000000000000009a>
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1cd0:  0x000000c00717a340  0x000000c00a2d24e8
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1ce0:  0x000000c009ab1d08  0x000000000040d976 <runtime.convI2I+0x0000000000000036>
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1cf0:  0x00000000051cce60  0x0000000000000000
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1d00:  0x000000c00717a300  0x000000c009ab1d80
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1d10:  0x00000000010e60a8 <k8s.io/apimachinery/pkg/util/framer.(*lengthDelimitedFrameReader).Read+0x0000000000000088>  0x00007f00299c29a8
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1d20:  0x000000c008d4bba0  0x000000c00a2d24e8
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1d30:  0x0000000000000004  0x0000000000000004
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1d40:  0x500000e8b7565000  0x004500082eacb756
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1d50:  0x06ff00401c023c00  0x1e0a44001e0adb64
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1d60:  0x507d010013804500  0x02a000000000ff2f
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1d70:  0x0402000057ca405b  0x020409030301b405
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1d80:  0x000071b8cc270a08  0x00000000010e0000 <k8s.io/client-go/pkg/apis/clientauthentication.(*ExecCredential).XXX_DiscardUnknown+0x0000000000000060>
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1d90: <0x000000c009758048  0x000000c009802000
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1da0:  0x0000000000000400  0x0000000000000400
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1db0: !0x0000000000000040 >0x0000000000000038
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1dc0:  0x000000000537e3c0  0x0000000000000000
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1dd0:  0x0000000000000000  0x00007f0151ceca68
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1de0:  0x0000000000000040  0x000000c000100000
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1df0:  0x000000c00617ca40  0x0000000000000000
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1e00:  0x0000000000000000  0x000000000040ff87 <runtime.newobject+0x0000000000000027>
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1e10:  0x0000000000000038  0x000000000537e3c0
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1e20:  0x000000c009ab1e01  0x000000c009ab1e88
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1e30:  0x00000000010e0a2f <k8s.io/client-go/rest/watch.(*Decoder).Decode+0x000000000000004f>  0x000000c00927cd20
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1e40:  0x000000c009ab1e68  0x000000000656a480
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1e50:  0x000000c00617ca40  0x0000000000001bc0
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1e60:  0x000000000963bb68  0x000000c009ab1eb8
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1e70:  0x000000c00617ca40  0x000000000963bb60
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1e80:  0x000000c009ab1eb8  0x000000c009ab1fb8
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1e90:  0x0000000001040e7c <k8s.io/apimachinery/pkg/watch.(*StreamWatcher).receive+0x00000000000000dc>  0x000000c0095681e0
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1ea0:  0x000000000963bb60  0x000000c002f23f1c
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: 0x000000c009ab1eb0:  0x000000000963bb70
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: fatal error: unknown caller pc
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime stack:
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime.throw({0x59e685b?, 0x94eddc0?})
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/panic.go:1047 +0x5d fp=0x7f0029835740 sp=0x7f0029835710 pc=0x43cebd
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime.gentraceback(0x30?, 0x6511658?, 0x0?, 0xc00717a340, 0x0, 0x0, 0x7fffffff, 0x7f0029835c78, 0x0?, 0x0)
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/traceback.go:270 +0x1bb0 fp=0x7f0029835a98 sp=0x7f0029835740 pc=0x466130
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime.scanstack(0xc00717a340, 0xc000087738)
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/mgcmark.go:804 +0x1f2 fp=0x7f0029835ca0 sp=0x7f0029835a98 pc=0x4229b2
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime.markroot.func1()
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/mgcmark.go:239 +0xb5 fp=0x7f0029835cf0 sp=0x7f0029835ca0 pc=0x4217b5
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime.markroot(0xc000087738, 0xe2d, 0x1)
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/mgcmark.go:213 +0x1a5 fp=0x7f0029835d90 sp=0x7f0029835cf0 pc=0x421465
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime.gcDrain(0xc000087738, 0x3)
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/mgcmark.go:1069 +0x39f fp=0x7f0029835df0 sp=0x7f0029835d90 pc=0x42355f
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime.gcBgMarkWorker.func2()
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/mgc.go:1348 +0xad fp=0x7f0029835e40 sp=0x7f0029835df0 pc=0x41fa2d
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime.systemstack()
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/asm_amd64.s:496 +0x49 fp=0x7f0029835e48 sp=0x7f0029835e40 pc=0x474029
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: goroutine 34 [GC worker (active)]:
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime.systemstack_switch()
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/asm_amd64.s:463 fp=0xc000508750 sp=0xc000508748 pc=0x473fc0
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime.gcBgMarkWorker()
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/mgc.go:1335 +0x205 fp=0xc0005087e0 sp=0xc000508750 pc=0x41f6c5
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime.goexit()
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/asm_amd64.s:1598 +0x1 fp=0xc0005087e8 sp=0xc0005087e0 pc=0x4761e1
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: created by runtime.gcBgMarkStartWorkers
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/mgc.go:1199 +0x25
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: goroutine 1 [chan receive, 1 minutes]:
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime.gopark(0xc003f68fc0?, 0xc00971ae10?, 0x20?, 0xb3?, 0xc000eb2d80?)
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/proc.go:381 +0xd6 fp=0xc0012ed798 sp=0xc0012ed778 pc=0x43fff6
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime.chanrecv(0xc000ad46c0, 0x0, 0x1)
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/chan.go:583 +0x49d fp=0xc0012ed828 sp=0xc0012ed798 pc=0x408a7d
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: runtime.chanrecv1(0x6596e48?, 0xc00056ca50?)
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/usr/local/go/src/runtime/chan.go:442 +0x18 fp=0xc0012ed850 sp=0xc0012ed828 pc=0x408578
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: github.com/k3s-io/k3s/pkg/agent.run({_, _}, {{0xc001d88000, 0x6a}, {0x0, 0x0}, {0x0, 0x0}, {0xc000cacd38, 0x16}, ...}, ...)
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/go/src/github.com/k3s-io/k3s/pkg/agent/run.go:177 +0xa45 fp=0xc0012edb38 sp=0xc0012ed850 pc=0x4146ce5
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: github.com/k3s-io/k3s/pkg/agent.Run({_, _}, {{0xc001d88000, 0x6a}, {0x0, 0x0}, {0x0, 0x0}, {0xc000cacd38, 0x16}, ...})
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/go/src/github.com/k3s-io/k3s/pkg/agent/run.go:274 +0x14e fp=0xc0012edda0 sp=0xc0012edb38 pc=0x414760e
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: github.com/k3s-io/k3s/pkg/cli/server.run(0xc000b5a580, 0x9634840, {0xc000ace9f8, 0x0, 0x0?}, {0xc000ace9f8, 0x0, 0x942aa20?})
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/go/src/github.com/k3s-io/k3s/pkg/cli/server/server.go:550 +0x3b88 fp=0xc0012ee9b8 sp=0xc0012edda0 pc=0x46f4128
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: github.com/k3s-io/k3s/pkg/cli/server.Run(0xc000b5a580?)
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/go/src/github.com/k3s-io/k3s/pkg/cli/server/server.go:41 +0x35 fp=0xc0012eea08 sp=0xc0012ee9b8 pc=0x46f0575
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: github.com/urfave/cli.HandleAction({0x4ee6d80?, 0x5e432b8?}, 0x6?)
Jun 17 20:04:11 10918364-mhr-ci-10918364-dut-2 k3s[37918]: #011/go/src/github.com/k3s-io/k3s/vendor/gith
```

**Comment by brandond** [comment]
This is a crash DEEP in a core go garbage collection function. I do not believe that this is something we can address here in K3s. The fact that you see this under load, and the resulting stack is in the go runtime garbage collection routine suggests that it is running out of memory and trying to force GC, and that is somehow running into memory corruption while doing so ( ref https://github.com/golang/go/issues/57550#issuecomment-1370153898).

How much memory does this node have? How much of that is available to k3s, when your workload is present? Have you tried profiling k3s while updating your application?

**Reply by sharrurashmi** [reply]
There was atleast 2GB of free memory available on the host as the other process are pretty much idle , the overall mem on machine was 16GB, Also each time we see a crash we have a different stack trace 
this is one of them 
d[1]: Removed slice libcontainer container kubepods-burstable-podbda01363_5b58_4269_9587_ee543543e822.slice.
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: I0621 12:59:46.901756  105840 job_controller.go:562] "enqueueing job" key="default/f5-api-service"
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: I0621 12:59:46.912702  105840 job_controller.go:562] "enqueueing job" key="default/f5-api-service"
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: unexpected fault address 0x0
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: fatal error: fault
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: [signal SIGSEGV: segmentation violation code=0x80 addr=0x0 pc=0x2571835]
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: goroutine 16036 [running]:
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: runtime.throw({0x59aebb9?, 0x0?})
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/usr/local/go/src/runtime/panic.go:1047 +0x5d fp=0xc013a037b8 sp=0xc013a03788 pc=0x43cebd
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: runtime.sigpanic()
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/usr/local/go/src/runtime/signal_unix.go:855 +0x28a fp=0xc013a03818 sp=0xc013a037b8 pc=0x454e6a
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: k8s.io/apimachinery/pkg/apis/meta/v1.(*Time).IsZero(...)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/time.go:63
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: k8s.io/kubernetes/pkg/kubelet/util.(*basicPodStartupLatencyTracker).ObservedPodOnWatch(0xc011c7baa0, 0xc01271e900, {0xc1957c30b66ebd06, 0xdecd59670, 0x962fba0})
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/k8s.io/kubernetes/pkg/kubelet/util/pod_startup_latency_tracker.go:80 +0x235 fp=0xc013a03978 sp=0xc013a03818 pc=0x2571835
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: k8s.io/kubernetes/pkg/kubelet/config.(*podStorage).merge.func1({0xc000bf6200?, 0x59ab3e0?, 0x3?}, 0x3?, 0x3ba2a90?)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/k8s.io/kubernetes/pkg/kubelet/config/config.go:248 +0x1be fp=0xc013a03a38 sp=0xc013a03978 pc=0x3c1479e
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: k8s.io/kubernetes/pkg/kubelet/config.(*podStorage).merge(0xc00746dd80, {0x59ab3e0, 0x3}, {0x54bb0e0?, 0xc00e3b4030})
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/k8s.io/kubernetes/pkg/kubelet/config/config.go:298 +0x424 fp=0xc013a03d20 sp=0xc013a03a38 pc=0x3c13604
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: k8s.io/kubernetes/pkg/kubelet/config.(*podStorage).Merge(0xc00746dd80, {0x59ab3e0, 0x3}, {0x54bb0e0, 0xc00e3b4030})
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/k8s.io/kubernetes/pkg/kubelet/config/config.go:166 +0x125 fp=0xc013a03e28 sp=0xc013a03d20 pc=0x3c12ca5
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: k8s.io/kubernetes/pkg/util/config.(*Mux).listen(...)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/k8s.io/kubernetes/pkg/util/config/config.go:85
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: k8s.io/kubernetes/pkg/util/config.(*Mux).ChannelWithContext.func1()
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/k8s.io/kubernetes/pkg/util/config/config.go:79 +0x71 fp=0xc013a03e90 sp=0xc013a03e28 pc=0x3c106d1
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:226 +0x3e fp=0xc013a03eb0 sp=0xc013a03e90 pc=0x10a765e
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc005104150?, {0x6552820, 0xc00cb206f0}, 0x1, 0x0)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:227 +0xb6 fp=0xc013a03f30 sp=0xc013a03eb0 pc=0x10a7516
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00b457fd0?, 0x0, 0x0, 0xc0?, 0xc00eeae920?)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:204 +0x89 fp=0xc013a03f80 sp=0xc013a03f30 pc=0x10a7409
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: k8s.io/apimachinery/pkg/util/wait.Until(0x2a3d840?, 0xc00c264cc0?, 0xc00b457fb8?)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:161 +0x25 fp=0xc013a03fb8 sp=0xc013a03f80 pc=0x10a7265
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: k8s.io/kubernetes/pkg/util/config.(*Mux).ChannelWithContext.func3()
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/k8s.io/kubernetes/pkg/util/config/config.go:79 +0x2c fp=0xc013a03fe0 sp=0xc013a03fb8 pc=0x3c1062c
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: runtime.goexit()
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/usr/local/go/src/runtime/asm_amd64.s:1598 +0x1 fp=0xc013a03fe8 sp=0xc013a03fe0 pc=0x4761e1
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: created by k8s.io/kubernetes/pkg/util/config.(*Mux).ChannelWithContext
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/k8s.io/kubernetes/pkg/util/config/config.go:79 +0x245
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: goroutine 1 [chan receive]:
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: runtime.gopark(0xc007de9c50?, 0xc00f1a43c0?, 0x20?, 0x21?, 0xc008540000?)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/usr/local/go/src/runtime/proc.go:381 +0xd6 fp=0xc000c9f798 sp=0xc000c9f778 pc=0x43fff6
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: runtime.chanrecv(0xc00078c6c0, 0x0, 0x1)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/usr/local/go/src/runtime/chan.go:583 +0x49d fp=0xc000c9f828 sp=0xc000c9f798 pc=0x408a7d
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: runtime.chanrecv1(0x6596e48?, 0xc000a5ab40?)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/usr/local/go/src/runtime/chan.go:442 +0x18 fp=0xc000c9f850 sp=0xc000c9f828 pc=0x408578
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: github.com/k3s-io/k3s/pkg/agent.run({_, _}, {{0xc003c068c0, 0x6a}, {0x0, 0x0}, {0x0, 0x0}, {0xc0005c71a0, 0x16}, ...}, ...)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/pkg/agent/run.go:177 +0xa45 fp=0xc000c9fb38 sp=0xc000c9f850 pc=0x4146ce5
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: github.com/k3s-io/k3s/pkg/agent.Run({_, _}, {{0xc003c068c0, 0x6a}, {0x0, 0x0}, {0x0, 0x0}, {0xc0005c71a0, 0x16}, ...})
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/pkg/agent/run.go:274 +0x14e fp=0xc000c9fda0 sp=0xc000c9fb38 pc=0x414760e
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: github.com/k3s-io/k3s/pkg/cli/server.run(0xc000e78c60, 0x9634840, {0xc0012309f8, 0x0, 0x0?}, {0xc0012309f8, 0x0, 0x942aa20?})
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/pkg/cli/server/server.go:550 +0x3b88 fp=0xc000ca09b8 sp=0xc000c9fda0 pc=0x46f4128
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: github.com/k3s-io/k3s/pkg/cli/server.Run(0xc000e78c60?)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/pkg/cli/server/server.go:41 +0x35 fp=0xc000ca0a08 sp=0xc000ca09b8 pc=0x46f0575
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: github.com/urfave/cli.HandleAction({0x4ee6d80?, 0x5e432b8?}, 0x6?)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/github.com/urfave/cli/app.go:524 +0x50 fp=0xc000ca0a20 sp=0xc000ca0a08 pc=0x257c6f0
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: github.com/urfave/cli.Command.Run({{0x59b29cc, 0x6}, {0x0, 0x0}, {0x0, 0x0, 0x0}, {0x59fac8e, 0x15}, {0xc0006bc768, ...}, ...}, ...)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/github.com/urfave/cli/command.go:175 +0x67b fp=0xc000ca0c50 sp=0xc000ca0a20 pc=0x257d53b
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: github.com/urfave/cli.(*App).Run(0xc000419c00, {0xc000b00a20, 0x11, 0x12})
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/vendor/github.com/urfave/cli/app.go:277 +0xb87 fp=0xc000ca1230 sp=0xc000ca0c50 pc=0x2579f07
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: main.main()
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/go/src/github.com/k3s-io/k3s/cmd/server/main.go:81 +0xc1e fp=0xc000ca1f80 sp=0xc000ca1230 pc=0x4a0779e
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: runtime.main()
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/usr/local/go/src/runtime/proc.go:250 +0x207 fp=0xc000ca1fe0 sp=0xc000ca1f80 pc=0x43fbc7
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: runtime.goexit()
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/usr/local/go/src/runtime/asm_amd64.s:1598 +0x1 fp=0xc000ca1fe8 sp=0xc000ca1fe0 pc=0x4761e1
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: goroutine 2 [force gc (idle)]:
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/usr/local/go/src/runtime/proc.go:381 +0xd6 fp=0xc0000a4fb0 sp=0xc0000a4f90 pc=0x43fff6
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: runtime.goparkunlock(...)
Jun 21 12:59:46 10965532-mhr-ci-10965532-dut-1 k3s[105840]: #011/usr/local/go/src/runtime/
